### ISSUE TITLE
saml.security.http.logout does not exist

### ISSUE THREAD

**Issue by paharasse** [hslavich/OneloginSamlBundle#104]
Hello,

I have this error when I activate logout config in my firewall : `Service "hslavich_onelogin_saml.saml_logout": Parent definition "saml.security.http.logout" does not exist.`
I have searched in my project and there is no place where saml.security.http.logout service is defined. Can you tell us how to configure it?
Thank you.

My firewall config :
```
firewalls:
        secured_area:
            pattern: ^/
            anonymous: true
            provider: fos_userbundle
            switch_user: { role: ROLE_ADMIN }
            saml:
                use_attribute_friendly_name: true
                user_factory: custo_user_provider
                check_path: saml_acs
                login_path: saml_login
            logout:
                path: saml_logout
```

**Comment by amalricBzh** [comment]
I had this error when upgrading from Symfony 3 et 4, I simply solved it with these operations:

- Backup your `hslavich_onelogin_saml` configuration
- `composer remove  hslavich/OneloginSamlBundle`
- `composer require hslavich/OneloginSamlBundle`
. Redo your `hslavich_onelogin_saml` configuration if needed (the `composer remove ...`may have remove it).

**Comment by rmdefi** [comment]
Same problem for a fresh install on a symfony 4 project (so I must stay with v1.x of this bundle for the time being).

I think the problem is with the createLogoutHandler() method in [SamlFactory](https://github.com/hslavich/OneloginSamlBundle/blob/1.x/DependencyInjection/Security/Factory/SamlFactory.php), which tries to define the service hslavich_onelogin_saml.saml_logout as a child of saml.security.http.logout, which is not defined anywhere else.

But that hslavich_onelogin_saml.saml_logout service is actually already defined in [services.yml](https://github.com/hslavich/OneloginSamlBundle/blob/1.x/Resources/config/services.yml)

For me, copying that definition in my own services.yml (which is loaded after the bundle factory, so can override what it did) fixed the problem.

I guess to fix the problem in v1.x, that setDefinition should be removed from createLogoutHandler(), but maybe it is needed to keep supporting symfony <4 ?

**Comment by a-menshchikov** [comment]
It seems to me that the problem is caused by accidentally copying the code from the pdias/SamlBundle package at the very beginning.

I removed the redundant code and released v1.5.3. I hope that should help.

I close this issue, but if the problem will not be solved, reopen it please.
